src/platform/utilities/api/index.js

```diff
@@ -145,15 +156,17 @@ export function apiRequest(resource, optionalSettings, success, error) {
         return data;
       }
 
-      if (environment.isProduction()) {
+      if (isProd) {
         const { pathname } = window.location;
 
         const shouldRedirectToSessionExpired =
           response.status === 401 &&
           !pathname.includes('auth/login/callback') &&
-          sessionStorage.getItem('shouldRedirectExpiredSession') === 'true';
+          sessionStorage.getItem('shouldRedirectExpiredSession') === 'true' &&
+          !pathname.includes('/terms-of-use/declined');
```

Comment:
Have you tested this manually to see if it works? I'm unable to get it to redirect properly when locally changing the `isProd = true`